dbms/src/Interpreters/Cluster.cpp

```diff
@@ -449,6 +450,10 @@ void Cluster::initMisc()
     }
 }
 
+std::unique_ptr<Cluster> Cluster::getClusterWithReplicasAsShards(const Settings & settings) const
+{
+    return std::unique_ptr<Cluster>{ new Cluster(settings, *this)};
```

Comment:
I'm remove the constructor as it is not intuitive at best. You have a copy constructor that changes the cluster behaviour by "flattening" all the shards. There is a high chance it will be miss-used by accident.

I believe there is one more problem/question, cross-replication setups where one node is part of multiple shards. Should we de-duplicate them instances by (host, port) pair in that case @alexey-milovidov? https://www.altinity.com/blog/2018/5/10/circular-replication-cluster-topology-in-clickhouse